com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ThreadsRequestHandler.java

```diff
@@ -45,36 +46,29 @@
     }
 
     @Override
-    public void handle(Command command, Arguments arguments, Response response, IDebugAdapterContext context) {
+    public CompletableFuture<Response> handle(Command command, Arguments arguments, Response response, IDebugAdapterContext context) {
         if (context.getDebugSession() == null) {
-            AdapterUtils.setErrorResponse(response, ErrorCode.EMPTY_DEBUG_SESSION, "Debug Session doesn't exist.");
-            return;
+            return AdapterUtils.createAsyncErrorResponse(response, ErrorCode.EMPTY_DEBUG_SESSION, "Debug Session doesn't exist.");
         }
         switch (command) {
             case THREADS:
-                this.threads((ThreadsArguments) arguments, response, context);
-                break;
+                return this.threads((ThreadsArguments) arguments, response, context);
             case STEPIN:
-                this.stepIn((StepInArguments) arguments, response, context);
-                break;
+                return this.stepIn((StepInArguments) arguments, response, context);
             case STEPOUT:
-                this.stepOut((StepOutArguments) arguments, response, context);
-                break;
+                return this.stepOut((StepOutArguments) arguments, response, context);
             case NEXT:
-                this.next((NextArguments) arguments, response, context);
-                break;
+                return this.next((NextArguments) arguments, response, context);
             case PAUSE:
-                this.pause((PauseArguments) arguments, response, context);
-                break;
+                return this.pause((PauseArguments) arguments, response, context);
             case CONTINUE:
-                this.resume((ContinueArguments) arguments, response, context);
-                break;
+                return this.resume((ContinueArguments) arguments, response, context);
             default:
-                return;
+                return null;
```

Comment:
Instead of returning null, try to complete the future with an exception by calling future.completeExceptionally

Reply:
Or return AdapterUtils.createAsyncErrorResponse

Reply:
nice.